fladgejt/webui/hodnotenia.py

```diff
@@ -53,3 +53,47 @@ def get_priemery(self, zapisny_list_key):
                           datum_vypoctu=row['priemerInfoDatum'])
                   for row in app.d.priemeryTable.all_rows()]
         return [result, None]
+
+    def get_priebezne_hodnotenia(self, zapisny_list_key):
+        try:
+            app = self._open_priebezne_hodnotenie_app(zapisny_list_key)
+        except CantOpenApplication:
+            return [[], "Priebežné hodnotenia pre tento zápisný list nie sú dostupné."]
+
+        # Ak uz bola aplikacia otvorena pred tymto volanim, stlacime "Refresh".
+        if getattr(app, 'freshly_opened', True):
+            app.freshly_opened = False
+        else:
+            app.d.refreshButton.click()
+
+        studium_key, akademicky_rok = decode_key(zapisny_list_key)
+
+        result = []
+        for i, row in enumerate(app.d.predmetySPriebeznymHodnotenimTable.all_rows()):
+            # Klikneme na predmet s priebeznym hodnotenim a pockame na nacitanie
+            with app.collect_operations() as ops:
+                app.d.predmetySPriebeznymHodnotenimTable.select(i)
+            if ops:
+                app.awaited_refresh_dialog(ops)
+
+            # Klikneme na button nacitatHodnotenie
+            with app.collect_operations() as ops:
+                app.d.nacitatHodnotenieButton.click()
+            if ops:
```

Comment:
Pre úplnosť, vysvetlenie: Závisí, čo sa stane. Keď AIS iba pošle nové dáta (napr. riadky v tabuľke) tak nič netreba. Keď AIS spraví niečo veľké, napr. otvorí nové okno, treba použiť `collect_operations` a skontrolovať, že sa stalo to, čo malo.
V niektorých dialógoch AIS robí refresh celého okna, čo sa ráta ako veľká vec. A v niektorých dialógoch to niekedy robí a niekedy nie, preto v iných súboroch je `if ops: app.awaited_refresh_dialog(ops)`.